src/v/cluster/persisted_stm.h

```diff
@@ -182,7 +197,7 @@ class persisted_stm
     model::offset _insync_offset;
     raft::consensus* _c;
     prefix_logger _log;
-    file_backed_stm_snapshot _snapshot_storage;
+    T _snapshot_storage;
```

Comment:
nit: let's be consistent in naming we called it backend so maybe `_snapshot_backend` ?